src/debug/di/shimstackwalk.cpp

```diff
@@ -1663,6 +1663,11 @@ HRESULT ShimChain::GetStackRange(CORDB_ADDRESS * pStart, CORDB_ADDRESS * pEnd)
         {
             *pEnd = PTR_TO_CORDB_ADDRESS(m_fpRoot.GetSPValue());
         }
+
+        if (m_fpRoot.GetSPValue() < CORDbgGetSP(&m_context))
```

Comment:
I would add some comments to this check with breadcrumbs that lead back to this issue. Failing this check presumably means we have a bug elsewhere in our code that got us into this bad state in the first place. Some day we might discover we need to understand and fix that root cause.